py4cast/lightning.py

```diff
@@ -701,6 +735,20 @@ def mask_tensor(self, x):
             ] = False
         return x * mask
 
+    def get_mask_on_nan(
+        self, target: NamedTensor, prediction: NamedTensor
+    ) -> torch.Tensor:
+        """
+        Returns a mask matching the nan values in target, same shape as the target.
+        Replaces the nan values by zeros in the target.
+        """
+        if self.mask_on_nan:
+            mask = ~torch.isnan(target.tensor)
+            target.tensor = torch.nan_to_num(target.tensor)
+            prediction.tensor = torch.nan_to_num(prediction.tensor)
+            return mask
+        return torch.ones_like(target.tensor)
```

Comment:
Does this function truly do what it's docstring say it does ?
What's the point of the unused `prediction: NamedTensor` parameter ?
```suggestion
    def get_mask_on_nan(self, target: NamedTensor) -> torch.Tensor:
        """Returns a mask matching the nan values in target, same shape as the target.
        Replaces the nan values by zeros in the target.
        """
        if self.mask_on_nan:
            return ~torch.isnan(target.tensor)
        target.tensor = torch.nan_to_num(target.tensor, nan=0)
        return torch.ones_like(target.tensor)
```